### PR TITLE
Support `Seq` aka `IEnumerable`

### PR DESCRIPTION
**Pull request by AlexeyRaga**
Sorry for bombarding you with PRs, but `seq` is yet another type that is good to have support for

### CODE REVIEW

**Comment by TysonMN** [comment]
Not a problem at all!  I am grateful for the contributions.

Can you split your one commit into two: the tests in the first commit and the fix in the second commit?

**Comment by AlexeyRaga** [comment]
Sure, done.

File: src/Hedgehog.Experimental.CSharp.Tests/GenTests.cs [hunk 1]
```diff
@@ -52,4 +53,18 @@ public void ShouldAddElementToList()
...
+  [Fact]
+  public void ShouldSupportIEnumerable() =>
+    GenX.auto<IEnumerable<int>>()
+      .Sample(1, 5)
+      .Should()
+      .AllSatisfy(x => x.Should().NotBeNull());
+
+  [Fact]
+  public void ShouldSupportList() =>
+    GenX.auto<List<int>>()
```
Comment (TysonMN) [inline src/Hedgehog.Experimental.CSharp.Tests/GenTests.cs:66]: I think the equivalent type in F# is `ResizeArray<int>`.

Can you check if the equivalent test fails for F#?

If it does, can you add it to your first commit?

**Review by TysonMN — CHANGES_REQUESTED** [review]

File: src/Hedgehog.Experimental.CSharp.Tests/GenTests.cs (hunk 1 above)
Comment (AlexeyRaga) [reply src/Hedgehog.Experimental.CSharp.Tests/GenTests.cs:66]: It does not fail, I was just checking that `Enumerable` pattern doesn't pick up anything but `IEnumerable`.

Comment (TysonMN) [reply src/Hedgehog.Experimental.CSharp.Tests/GenTests.cs:66]: Ok.  Then please split your first commit into two: the first commit containing this passing test and the second commit containing your two failing tests.

**Review by jeffreysmooths — APPROVED** [review]

File: src/Hedgehog.Experimental.CSharp.Tests/GenTests.cs (hunk 1 above)
Comment (AlexeyRaga) [reply src/Hedgehog.Experimental.CSharp.Tests/GenTests.cs:66]: I'll rather remove the tests that I added, these splitting/merging are getting involved.

Comment (TysonMN) [reply src/Hedgehog.Experimental.CSharp.Tests/GenTests.cs]: Thanks for doing that (and for your patience :) ).

**Comment by TysonMN** [comment]
First included in [version 0.7.0](https://www.nuget.org/packages/Hedgehog.Experimental/0.7.0)

**Review by jeffreysmooths — COMMENTED** [review]
Good

**Pull request merged**